From 14a5b665c99406442679ebf086ee97aab83765c0 Mon Sep 17 00:00:00 2001
From: Haohui Mai <wheat9@apache.org>
Date: Tue, 13 Oct 2015 13:07:57 -0700
Subject: [PATCH 1001/1023] HADOOP-12474. MiniKMS should use random ports for
 Jetty server by default. Contributed by Mingliang
 Liu.

(cherry picked from commit 950e8a459ea8ec3785f07ba67540eb6dac217ffd)

Change-Id: I77ef59a70e870aa4b09cbc8625845a71cba95d8b
---
 .../hadoop/crypto/key/kms/server/MiniKMS.java      |   12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/hadoop-common-project/hadoop-kms/src/test/java/org/apache/hadoop/crypto/key/kms/server/MiniKMS.java b/hadoop-common-project/hadoop-kms/src/test/java/org/apache/hadoop/crypto/key/kms/server/MiniKMS.java
index 4f802cc..eb640bb 100644
--- a/hadoop-common-project/hadoop-kms/src/test/java/org/apache/hadoop/crypto/key/kms/server/MiniKMS.java
+++ b/hadoop-common-project/hadoop-kms/src/test/java/org/apache/hadoop/crypto/key/kms/server/MiniKMS.java
@@ -47,19 +47,13 @@
   private static Server createJettyServer(String keyStore, String password, int inPort) {
     try {
       boolean ssl = keyStore != null;
-      InetAddress localhost = InetAddress.getByName("localhost");
       String host = "localhost";
-      ServerSocket ss = new ServerSocket((inPort < 0) ? 0 : inPort, 50, localhost);
-      int port = ss.getLocalPort();
-      ss.close();
-      Server server = new Server(0);
+      Server server = new Server(inPort);
       if (!ssl) {
         server.getConnectors()[0].setHost(host);
-        server.getConnectors()[0].setPort(port);
       } else {
         SslSocketConnector c = new SslSocketConnectorSecure();
         c.setHost(host);
-        c.setPort(port);
         c.setNeedClientAuth(false);
         c.setKeystore(keyStore);
         c.setKeystoreType("jks");
@@ -80,7 +74,7 @@ private static URL getJettyURL(Server server) {
       String scheme = (ssl) ? "https" : "http";
       return new URL(scheme + "://" +
           server.getConnectors()[0].getHost() + ":" +
-          server.getConnectors()[0].getPort());
+          server.getConnectors()[0].getLocalPort());
     } catch (MalformedURLException ex) {
       throw new RuntimeException("It should never happen, " + ex.getMessage(),
           ex);
@@ -92,7 +86,7 @@ private static URL getJettyURL(Server server) {
     private String log4jConfFile;
     private File keyStoreFile;
     private String keyStorePassword;
-    private int inPort = -1;
+    private int inPort;
 
     public Builder() {
       kmsConfDir = new File("target/test-classes").getAbsoluteFile();
-- 
1.7.9.5

